tests/end-to-end/director/labeling_api_test.go

```diff
@@ -0,0 +1,508 @@
+
+		t.Log("Set label on application")
+		setLabelRequest := fixSetApplicationLabelRequest(app.ID, labelKey, appLabel)
+		label := graphql.Label{}
+
+		err = tc.RunQuery(ctx, setLabelRequest, &label)
+		defer deleteLabelDefinition(t, ctx, labelKey, false)
+		defer deleteApplicationLabel(t, ctx, app.ID, labelKey)
+
+		var invalidSchema interface{} = invalidJsonSchema
+		labelDefinitionInput = graphql.LabelDefinitionInput{
+			Key:    labelKey,
+			Schema: &invalidSchema,
+		}
+
+		ldInputGql, err := tc.graphqlizer.LabelDefinitionInputToGQL(labelDefinitionInput)
+		require.NoError(t, err)
+
+		updateLabelDefinitionReq := fixUpdateLabelDefinitionRequest(ldInputGql)
+
+		// WHEN
+		t.Log("Try to edit LabelDefinition with incompatible data")
+		err = tc.RunQuery(context.Background(), updateLabelDefinitionReq, nil)
+
+		//THEN
+		require.Error(t, err)
+		errString := fmt.Sprintf("graphql: while updating label definition: label with key \"%s\" is not valid against new schema for Application with ID \"%s\": %s: Invalid type. Expected: integer, given: string", labelKey, app.ID, labelKey)
+		assert.EqualError(t, err, errString)
+	})
+
+	t.Run("Edit LabelDefinition with compatible data", func(t *testing.T) {
+		createLabelDefinitionRequest := fixCreateLabelDefinitionRequest(labelDefinitionInputGQL)
+		labelDefinition := graphql.LabelDefinition{}
+
+		t.Log("Create application")
+		app := createRandomApplication(t, ctx, "app")
+		defer deleteApplication(t, app.ID)
+
+		t.Log("Create label definition")
+		err = tc.RunQuery(ctx, createLabelDefinitionRequest, &labelDefinition)
+		require.NoError(t, err)
+
+		t.Log("Set label on application")
+		setLabelRequest := fixSetApplicationLabelRequest(app.ID, labelKey, appLabel)
+		label := graphql.Label{}
+
+		err = tc.RunQuery(ctx, setLabelRequest, &label)
+		defer deleteLabelDefinition(t, ctx, labelKey, false)
+		defer deleteApplicationLabel(t, ctx, app.ID, labelKey)
+
+		var newSchema interface{} = newValidJsonSchema
+		labelDefinitionInput = graphql.LabelDefinitionInput{
+			Key:    labelKey,
+			Schema: &newSchema,
+		}
+
+		ldInputGql, err := tc.graphqlizer.LabelDefinitionInputToGQL(labelDefinitionInput)
+		require.NoError(t, err)
+
+		updateLabelDefinitionReq := fixUpdateLabelDefinitionRequest(ldInputGql)
+
+		// WHEN
+		t.Log("Edit LabelDefinition with compatible data")
+		err = tc.RunQuery(context.Background(), updateLabelDefinitionReq, &labelDefinition)
+
+		//THEN
+		require.NoError(t, err)
+
+		schemaVal, ok := (*labelDefinition.Schema).(map[string]interface{})
+		require.True(t, ok)
+		actualProperties, ok := schemaVal["properties"].(map[string]interface{})
+		require.True(t, ok)
+
+		expectedProperties, ok := newValidJsonSchema["properties"].(map[string]interface{})
+		require.True(t, ok)
+
+		assert.Equal(t, expectedProperties, actualProperties)
+		saveQueryInExamples(t, updateLabelDefinitionReq.Query(), "update label definition")
+	})
+}
+
+func TestCheckIfLabelScenarioWasCreated(t *testing.T) {
```

Comment:
Let's be consistent - `TestCreateScenariosLabel`

Reply:
renamed